src/pybind/mgr/dashboard/services/orchestrator.py

```diff
@@ -122,6 +123,7 @@ def reload(self, service_type, service_ids):
     @wait_api_result
     def apply(self, service_spec: Dict) -> OrchResult[List[str]]:
         spec = ServiceSpec.from_json(service_spec)
+        spec.validate()
```

Comment:
@sebastian-philipp Just an observation: Wouldn't be better to call `validate()` as last step inside `ServiceSpec.from_json()` (and maybe whe ncreating new instances)
so we don't have to explictly invoke the validation?

Reply:
That was the case previously, yeah and I'm wondering if my new approach actually makes sense. My problem is: `ceph orch ls` needs to be able to return invalid ServiceSpecs, and I'm a bit out of ideas how to do that

Reply:
Does it make sense to add a flag like `allow_invalid=False` for those particular cases?

Reply:
seems like this would reduce the amount of exception handling .. 

we might also want to raise a HEALTH_WARN and/or denote this via `ceph orch ls` to help users locate which specs need attention

Reply:
> we might also want to raise a HEALTH_WARN and/or denote this via ceph orch ls to help users locate which specs need attention

Actually I'm :-1: here. In case we have invalid specs in our store, we should bail out and let users report a bug instead of adding a new kind of health warning.